packages/kwenta/src/Kwenta.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  SMART_MARGIN_V3,
+  SMART_MARGIN_V3_COMMIT_ORDER_ABI,
+  USDC,
+} from './constants.js'
+
+import {
+  type TransactionFilter,
+  type OptionsActionParams,
+  compressJson,
+} from '@rabbitholegg/questdk'
+import { type Address } from 'viem'
+import { Chains } from '@rabbitholegg/questdk-plugin-utils'
+
+/*
+ * Function templates for handling various blockchain action types.
+ * It's adaptable for actions defined in ActionParams: Bridge, Swap, Stake, Mint, Delegate, Quest, Etc.
+ * Duplicate and customize for each specific action type.
+ * If you wish to use a different action other than swap, import one of the ActionParams types
+ * from @rabbitholegg/questdk (ie: SwapActionParams) and change the function below to use
+ * the action params you wish to use.
+ */
+
+export const options = async (
+  _params: OptionsActionParams,
+): Promise<TransactionFilter> => {
+  // the ActionParams for this function are populated in the Boost Manager when the actual Boost is launched.
+
+  // In this function you should load the ABI, and translate any ActionParams into the input object defined below
+  // which should match the parameter names in the transaction
+
+  // You can also use the boostdk filter system to support operators on parameters, for example, greater than
+
+  // We always want to return a compressed JSON object which we'll transform into a TransactionFilter
+  const { chainId, recipient, amount } = _params
+
+  return compressJson({
+    chainId,
+    to: SMART_MARGIN_V3,
+    from: recipient,
+    input: {
+      $abiAbstract: SMART_MARGIN_V3_COMMIT_ORDER_ABI,
+      _sizeDelta: amount,
```

Comment:
We will also need to add a way to determine the tokenAddress, and orderType. This data should hopefully be encoded in the transaction details.

tokenAddress for example will likely be able to be determined from the _perpsMarketId

You will need to find out the mapping between marketId and tokens. marketId of 100 seems to be ETH-USD for example.

Reply:
100 = ETH-USD
200 = BTC-USD (you can use wBTC for this)

Reply:
@mmackz You mean that only these two pairs should be used, don't you?

Reply:
These are the only available pairs.

![CleanShot 2024-03-29 at 08 06 02@2x](https://github.com/rabbitholegg/questdk-plugins/assets/62824345/55ee35be-76b6-49e5-8500-b8b0fc9a451f)